contracts/HATVault.sol

```diff
@@ -213,6 +214,8 @@ contract HATVault is IHATVault, ERC4626Upgradeable, OwnableUpgradeable, Reentran
             revert NotSafetyPeriod();
         if (_bountyPercentage > maxBounty)
             revert BountyPercentageHigherThanMaxBounty();
+        if (maxBounty == HUNDRED_PERCENT && _bountyPercentage != HUNDRED_PERCENT)
```

Comment:
I am not sure if this is the logic we want. Seems to me a cleaner API is if "maxBounty" continus to mean "the maxium amount you can pay out"  and we enforce a gap here something like
```suggestion
        if (maxBounty == HUNDRED_PERCENT && (_bountyPercentage > MAX_BOUNTY_LIMIT && _bountyPercentage != HUNDRED_PERCENT)
```